features/cellular/framework/API/CellularDevice.h

```diff
@@ -97,6 +115,14 @@ class CellularDevice {
      */
     virtual void close_information() = 0;
 
+    /** Closes all the opened interfaces by deleting deleting this instance.
+     */
+    virtual void close_and_delete() = 0;
+
+    /** Closes all interfaces.
+     */
+    virtual void close_all_interfaces() = 0;
```

Comment:
I guess that get_instance, close_and_delete and close_all_interfaces should not be public - due to it's not up to an application to handle Device lifecycle. Instead, the device should be created on the first open_interface and deleted when the last of its interfaces is closed. That is, open_interfaces needs to be reference counted and call internally get_instance/delete.

Reply:
Removed methods. Agreed that there is only one CellularDevice and it can't be deleted.